### PR TITLE
Http static node

### PR DESCRIPTION
**Pull request by ceolin**
- Add new node to serve static files
- Fix the remove_dir function on impl-microhttpdc

### CODE REVIEW

File: src/modules/flow/http-server/http-server.c [hunk 1]
```diff
@@ -497,4 +497,77 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
...
+static int
+static_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_server_static_options *opts =
+        (struct sol_flow_node_type_http_server_static_options *)options;
+
+    if (!server) {
+        server = sol_http_server_new(HTTP_SERVER_PORT);
+        SOL_NULL_CHECK(server, -1);
```
Comment (barbieri) [inline src/modules/flow/http-server/http-server.c]: should use some -errno, maybe `-ENOMEM` or something like that.

also, would be nice to have this and the init_count as a function that is common to all node types.

File: src/modules/flow/http-server/http-server.c [hunk 2]
```diff
@@ -497,4 +497,77 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
...
+    }
+    init_count++;
+
+    mdata->path = strdup(opts->path);
+    SOL_NULL_CHECK_GOTO(mdata->path, err);
+
+    mdata->value.b = opts->start;
+
+    if (opts->start)
+        return sol_http_server_add_dir(server, mdata->path);
```
Comment (barbieri) [inline src/modules/flow/http-server/http-server.c]: there is a missing argument here. This directory should be exposed in the server at some other directory. Like the directory `/srv/http` being exposed as `/static` in the server.

Comment (barbieri) [inline src/modules/flow/http-server/http-server.c]: also, similar to request handlers that return a handler, we could return a handle from `sol_http_server_add_dir()`, using that to remove/destroy.

File: src/modules/flow/http-server/http-server.c [hunk 3]
```diff
@@ -497,4 +497,77 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
...
+    struct http_data *mdata = data;
+
+    if (mdata->value.b)
+        sol_http_server_remove_dir(server, mdata->path);
+
+    init_count--;
+    if (!init_count) {
+        sol_http_server_del(server);
+        server = NULL;
+    }
```
Comment (barbieri) [inline src/modules/flow/http-server/http-server.c]: this block should be a common function as well.

**Comment by ceolin** [comment]
merged with the suggestion, api change will come in another pullreq.Thanks

**Pull request closed without merging**